okhttp/src/test/java/okhttp3/internal/ws/WebSocketHttpTest.java

```diff
@@ -84,6 +84,7 @@
     clientListener.assertExhausted();
 
     // TODO: assert all connections are released once leaks are fixed
+    clientTestRule.abandonClient(client);
```

Comment:
The comment above makes it clear why this test was failing...

The rule is still worth using here to standardise the test client creation, but we just don't end up with clean client instances.